src/solver/variable/include/antares/solver/variable/economy/dispatchableGeneration.h

```diff
@@ -169,84 +125,95 @@ class DispatchableGeneration
     };
 
 public:
+    DispatchableGeneration():
+        pValuesForTheCurrentYear(nullptr)
+    {
+    }
+
     ~DispatchableGeneration()
     {
+        for (unsigned int numSpace = 0; numSpace < pNbYearsParallel; numSpace++)
+        {
+            delete[] pValuesForTheCurrentYear[numSpace];
+        }
         delete[] pValuesForTheCurrentYear;
     }
 
-    void initializeFromStudy(Data::Study& study)
+    void initializeFromArea(Data::Study* study, Data::Area* area)
     {
-        pNbYearsParallel = study.maxNbYearsInParallel;
+        pNbYearsParallel = study->maxNbYearsInParallel;
+        pValuesForTheCurrentYear = new VCardType::IntermediateValuesBaseType[pNbYearsParallel];
 
-        InitializeResultsFromStudy(AncestorType::pResults, study);
+        // Building the vector of group names the clusters belong to.
+        std::set<std::string> names;
+        for (const auto& cluster: area->thermal.list.each_enabled())
+        {
+            names.insert(cluster->getGroup());
+        }
+        groupNames_ = {names.begin(), names.end()};
+        groupToNumbers_ = Utils::giveNumbersToStrings(groupNames_);
 
-        pValuesForTheCurrentYear = new VCardType::IntermediateValuesBaseType[pNbYearsParallel];
-        for (unsigned int numSpace = 0; numSpace < pNbYearsParallel; ++numSpace)
+        nbColumns_ = groupNames_.size();
+
+        if (nbColumns_)
         {
-            for (unsigned int i = 0; i != VCardType::columnCount; ++i)
+            AncestorType::pResults.resize(nbColumns_);
+
+            for (unsigned int numSpace = 0; numSpace < pNbYearsParallel; numSpace++)
             {
-                pValuesForTheCurrentYear[numSpace][i].initializeFromStudy(study);
+                pValuesForTheCurrentYear[numSpace] = new VCardType::IntermediateValuesDeepType
+                  [nbColumns_];
```

Comment:
Can you please see this https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/2595 and remove dynamic allocations ?

Probably you'll have merge conflicts